cpp/include/raft/linalg/reduce.cuh

```diff
@@ -75,6 +78,98 @@ void reduce(OutType* dots,
     dots, data, D, N, init, rowMajor, alongRows, stream, inplace, main_op, reduce_op, final_op);
 }
 
+/**
+ * @defgroup reduction Reduction Along Requested Dimension
+ * @{
+ */
+
+/**
+ * @brief Compute reduction of the input matrix along the requested dimension
```

Comment:
The descriptions for the new functions could be made more verbose and explicit. The way I think about it is that a user should be able to look at the docs for this function and see right in the description 1) what exactly it's going to do to their data. The parameter/argument document should be expand upon that. 

Also, now that we have a nice clean API, usage examples are super helpful as well (like CUB and Thrust), it's really useful to have a copy/paste example that just works, vs having to figure out all the details before making the call.